### PR TITLE
zktx: add status and storage layer abstraction and helper functions

### PR DESCRIPTION
**Pull request by drmingdrmer**

### CODE REVIEW

File: zktx/README.md [hunk 1]
```diff
@@ -46,6 +47,201 @@ def get(self):
...
+
+-   `key`:
+    record key.
+
+-   `value`:
+    record value.
+
+**return**:
+a `bool` indicates if change has been made to underlying storage.
+Normal it is `False` if a higher txid has already applied.
```
Comment (liubaohai) [inline zktx/README.md]: has been

File: zktx/README.md [hunk 2]
```diff
@@ -46,6 +47,201 @@ def get(self):
...
+
+It requires 1 accessor method: `self.record.get(key)`.
+
+**arguments**:
+
+-   `key`:
+    specifies the `key` of the record.
+
+**return**:
+a dict in form of `{"txid": ..., "value": ...}` and a implementation defined version.
```
Comment (liubaohai) [inline zktx/README.md]: an implementation

File: zktx/README.md [hunk 3]
```diff
@@ -46,6 +47,201 @@ def get(self):
...
+It should release the lock identified by `key`, if and only if the lock is held
+by `txid`
+
+> This way only the lock holder could release the lock, without the need to
+> know if it has already acquired the lock.
+> This makes the recovery of a tx processor very easy:
+> A recovered process only need to know the `txid` but not the locking
+> informations.
+
+It should returns 3 element `tuple`:
```
Comment (liubaohai) [inline zktx/README.md]: elements

File: zktx/README.md [hunk 4]
```diff
@@ -46,6 +47,201 @@ def get(self):
...
+-   A `bool` indicates if the lock is previously locked by any one.
+
+-   A `txid` indicates current lock holder.
+    If no one has been locking it, it is the passed in `txid`
+
+-   A 3rd value indicates lock stat(not used yet).
+
+### TXStorage helper methods
+
+There are also two 3 methods an TX engine requires, which are already provided
```
[the remark this answers is not in the document]
Comment (drmingdrmer) [reply zktx/README.md]: 数字忘改了...是3...

File: zktx/README.md (hunk 3 above)
Comment (drmingdrmer) [reply zktx/README.md]: 这个就是单数应该...

**Pull request merged**